### PR TITLE
Checkout fe

### PR DESCRIPTION
**Pull request by TarasKun**
add modal
add fondy
refactoring

### CODE REVIEW

File: src/containers/checkout/delivery/delivery-type/mail-services/nova-poshta/nova-poshta-bottom.js [hunk 1]
```diff
@@ -23,22 +28,28 @@ const NovaPoshtaBottom = ({ city }) => {
 
   useEffect(() => {
     city && dispatch(getNovaPoshtaWarehouse(city));
-    setInputValue('');
-  }, [dispatch, city]);
+    setDepartmentValue('');
+  }, [dispatch, city, setDepartmentValue]);
 
-  const [inputValue, setInputValue] = useState('');
+  // const [departmentValue, setDepartmentValue] = useState('');
```
Comment (moran711) [inline src/containers/checkout/delivery/delivery-type/mail-services/nova-poshta/nova-poshta-bottom.js:34]: Please delete this commented line

Comment (TarasKun) [reply src/containers/checkout/delivery/delivery-type/mail-services/nova-poshta/nova-poshta-bottom.js]: fixed

**Review by DmytroDidukh — APPROVED** [review]
Great job!

File: src/containers/checkout/delivery/delivery-type/delivery-type.js [hunk 2]
```diff
@@ -1,32 +1,75 @@
...
 
   const [city, setCity] = useState('');
-  const dispatch = useDispatch();
-
+  const [departmentValue, setDepartmentValue] = useState('');
+  const [streetValue, setStreetValue] = useState('');
+  const [buildValue, setBuildValue] = useState('');
+  const [apartmentValue, setApartmentValue] = useState('');
+  const [totalPrice, setTotalPrice] = useState(0);
   const { cities } = useSelector(({ Checkout }) => ({
```
[the remark this answers is not in the document]
Comment (TarasKun) [reply src/containers/checkout/delivery/delivery-type/delivery-type.js]: fixed

**Review by moran711 — APPROVED** [review]

File: src/containers/checkout/delivery/delivery-type/modal/modal.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    <div className={style.orderFormModal}>
+      <div className={style.modalTitle}>
+        <span className={style.modalOrderTitle}>
+          {CHECKOUT_TITLES[language].orderNumber}
+        </span>
+        <span className={style.modalOrderSubTitle}>{fondyData.orderID}</span>
+      </div>
+      <div>
+        <span className={style.modalSubTitle}>
+          {CHECKOUT_TEXT_FIELDS[language].firstName}:{' '}
```
Comment (TorskyM) [inline src/containers/checkout/delivery/delivery-type/modal/modal.js:42]: Use string interpolation here

File: src/containers/checkout/delivery/delivery-type/modal/modal.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        </span>
+        <span className={style.modalData}>
+          {personalData.firstName} {personalData.lastName}
+        </span>
+      </div>
+      <div>
+        <span className={style.modalSubTitle}>
+          {CHECKOUT_TEXT_FIELDS[language].contactPhoneNumber}:{' '}
```
Comment (TorskyM) [inline src/containers/checkout/delivery/delivery-type/modal/modal.js:50]: Same

File: src/redux/checkout/checkout.sagas.js [hunk 5]
```diff
@@ -22,6 +25,28 @@ function* handleErrors(error) {
   yield put(push('/error-page'));
 }
 
+export function* handleFondyUrl({ payload }) {
+  try {
+    const dataFromFondy = yield call(
+      getItems,
+      `query{
+              getPaymentCheckout(data: {
+                orderId: "${payload.orderID}",
```
Comment (TorskyM) [inline src/redux/checkout/checkout.sagas.js:34]: Why do you need `" "` here?

File: src/containers/checkout/delivery/delivery-type/mail-services/courier/courier-bottom.js [hunk 6]
```diff
@@ -18,14 +27,18 @@ const CurrierBottom = ({ cityForNovaPoshtaBottom }) => {
     })
   );
 
-  const [inputValue, setInputValue] = useState('');
+  useEffect(() => {
+    streetValue && apartmentValue && buildValue
+      ? handleDeliveryTypeValidator(true)
+      : handleDeliveryTypeValidator(false);
+  }, [streetValue, handleDeliveryTypeValidator, apartmentValue, buildValue]);
```
Comment (ahd3r) [inline src/containers/checkout/delivery/delivery-type/mail-services/courier/courier-bottom.js:34]: Why `handleDeliveryTypeValidator` in dependencies?

File: src/containers/checkout/delivery/delivery-type/mail-services/nova-poshta/nova-poshta-bottom.js [hunk 7]
```diff
@@ -23,22 +28,26 @@ const NovaPoshtaBottom = ({ city }) => {
 
   useEffect(() => {
     city && dispatch(getNovaPoshtaWarehouse(city));
-    setInputValue('');
-  }, [dispatch, city]);
+    setDepartmentValue('');
+  }, [dispatch, city, setDepartmentValue]);
```
Comment (ahd3r) [inline src/containers/checkout/delivery/delivery-type/mail-services/nova-poshta/nova-poshta-bottom.js:32]: Why you have a setter (`setDepartmentValue`) in dependencies, is there any sense to have it?

File: src/containers/checkout/delivery/delivery-type/mail-services/nova-poshta/nova-poshta-bottom.js [hunk 8]
```diff
@@ -23,22 +28,26 @@ const NovaPoshtaBottom = ({ city }) => {
...
 
-  const [inputValue, setInputValue] = useState('');
+  useEffect(() => {
+    departmentValue
+      ? handleDeliveryTypeValidator(true)
+      : handleDeliveryTypeValidator(false);
+  }, [departmentValue, handleDeliveryTypeValidator]);
```
Comment (ahd3r) [inline src/containers/checkout/delivery/delivery-type/mail-services/nova-poshta/nova-poshta-bottom.js:38]: The same with `handleDeliveryTypeValidator`?

File: src/containers/checkout/delivery/delivery-type/modal/modal.js [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+  const { language, fondy } = useSelector(({ Language, Checkout }) => ({
+    language: Language.language,
+    fondy: Checkout.fondyData
+  }));
+
+  const style = useStyles();
+
+  useEffect(() => {
+    shouldValidate && allFieldsValidated && setOpenModal(true);
+  }, [shouldValidate, allFieldsValidated, setOpenModal]);
```
Comment (ahd3r) [inline src/containers/checkout/delivery/delivery-type/modal/modal.js:30]: Setter in dependencies?

File: src/containers/checkout/delivery/delivery-type/modal/modal.js (hunk 4 above)
Comment (TarasKun) [reply src/containers/checkout/delivery/delivery-type/modal/modal.js]: fixed

File: src/containers/checkout/delivery/delivery-type/modal/modal.js (hunk 3 above)
Comment (TarasKun) [reply src/containers/checkout/delivery/delivery-type/modal/modal.js]: fixed

**Review by ahd3r — APPROVED** [review]

**Pull request merged**